### ISSUE TITLE
parse_datetime_duration doesn't understand hours with no minutes, anymore

### ISSUE THREAD

**Issue by menozero** [stsc/DateTime-Format-Natural#19]
Hi. I've been using this module for more than ten years (last bug report: 2009!). I've noticed that the new version (1.13-1 on Ubuntu 22.04, but I also tested the latest with perlbrew) is not-so-compatible with the older one.

I've used it with lines like:

```
2022-09-01 14:35 to 18
```

but now that doesn't work: `Invalid day of month (day = 31 - month = 9 - year = 2022)`
Using `18:00` still works. So, no big deal, I can live with it, just to let you know.

Thanks for your work!
Giovanni

Small test, the first one passes, the second fails:

```
diff --git a/t/04-parse_durations.t b/t/04-parse_durations.t
index 1243d62..a1709f6 100755
--- a/t/04-parse_durations.t
+++ b/t/04-parse_durations.t
@@ -13,6 +13,8 @@ my @absolute = (
     { 'monday to friday' => [ '20.11.2006 00:00:00', '24.11.2006 00:00:00' ] },
     { 'march to august'  => [ '01.03.2006 00:00:00', '01.08.2006 00:00:00' ] },
     { '1999 to 2006'     => [ '01.01.1999 00:00:00', '01.01.2006 00:00:00' ] },
+    { '2022-09-01 14:35 to 18:00' => [ '01.09.2022 14:35:00', '01.09.2022 18:00:00' ] },
+    { '2022-09-01 14:35 to 18' => [ '01.09.2022 14:35:00', '01.09.2022 18:00:00' ] },
 );
 
 my @combined = (
```

**Comment by stsc** [comment]
Do you by any chance know when it stopped working?

**Comment by menozero** [comment]
Not precisely.

Strangely, in August it worked (or, I didn't notice it), and I was already using the same Ubuntu 22.04 with system module 1.13-1. (Previous Ubuntu LTS was 1.08-1.)

I have the feeling that it works with different dates (different months?)... testing...

Yes, it works with `2022-08-03 14 to 18` (and `2022-09-01 14 to 18`), but not with `2022-09-01 14:35 to 18`.

So, it's less general. (And more confusing, sorry.)

I may try to test it on different versions, but I see that the repository is not tagged with the releases.

**Comment by stsc** [comment]
The difficulty is that `::Duration::Checks` distinguishes between type categories and acts upon that :-).

I managed to get a small patch working by rewriting the duration string before it gets parsed.

Please check the attached patch if it goes in the right direction for you!

[dtfn.patch.txt](https://github.com/stsc/DateTime-Format-Natural/files/11715514/dtfn.patch.txt) `(based on v1.16)`

**Comment by stsc** [comment]
Enhanced patch now available in stable `v1.17`.

Closing.
